Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -901,24 +901,24 @@ public Modes getModes() {
     }
 
     @Override
-    public boolean canChooseTarget(Game game) {
+    public boolean canChooseTarget(Game game, UUID playerId) {
         if (this instanceof SpellAbility) {
             if (SpellAbilityType.SPLIT_FUSED.equals(((SpellAbility) this).getSpellAbilityType())) {
                 Card card = game.getCard(getSourceId());
                 if (card != null) {
-                    return canChooseTargetAbility(((SplitCard) card).getLeftHalfCard().getSpellAbility(), game, getControllerId())
-                            && canChooseTargetAbility(((SplitCard) card).getRightHalfCard().getSpellAbility(), game, getControllerId());
+                    return canChooseTargetAbility(((SplitCard) card).getLeftHalfCard().getSpellAbility(), game, playerId)
+                            && canChooseTargetAbility(((SplitCard) card).getRightHalfCard().getSpellAbility(), game, playerId);
                 }
                 return false;
             }
         }
-        return canChooseTargetAbility(this, game, getControllerId());
+        return canChooseTargetAbility(this, game, playerId);
     }
 
     private static boolean canChooseTargetAbility(Ability ability, Game game, UUID controllerId) {
         int found = 0;
         for (Mode mode : ability.getModes().values()) {
-            if (mode.getTargets().canChoose(ability.getSourceId(), ability.getControllerId(), game)) {
+            if (mode.getTargets().canChoose(ability.getSourceId(), controllerId, game)) {
```

Comment:
Hmm... Can you look at change history from cba7a510ea8722a0a8c60e8b61f4f32bc8816f20
> Fixed a bug if casting split cards from other players e.g with Mindclaw Shaman (fixes #3867).

Is it working with new changes?

Reply:
It works just like it did before but according to this you're not supposed to be able to cast a fused card from the opponent's hand:  https://magicjudge.tumblr.com/post/144389585306/does-mindclaw-shaman-allow-you-to-cast-both-halves

> 
> Sadly not.
> 
> You can only ever cast both halves of a split card with Fuse when you’re casting it from YOUR hand. Casting it from another player’s hand is just like casting it from anywhere that’s not your hand: You can only cast one half of it.

Reply:
Then all fine.